examples/Microsoft.Spark.CSharp.Examples/MachineLearning/SentimentStream/README.md

```diff
@@ -0,0 +1,229 @@
+# Sentiment Analysis with Big Data
+
+In this sample, you'll see how to use [.NET for Apache Spark](https://dotnet.microsoft.com/apps/data/spark) 
+and [ML.NET](https://dotnet.microsoft.com/apps/machinelearning-ai/ml-dotnet) to determine if 
+statements are positive or negative, a task known as **sentiment analysis**.
+
+## Problem
+
+Our goal here is to determine if statements typed into a console are positive or negative. We'll be using .NET for Apache Spark to process statements in real-time and ML.NET to perform **binary classification** since categorizing text involves choosing one of two groups: positive or negative. You can read more about the problem through the [ML.NET documentation](https://docs.microsoft.com/en-us/dotnet/machine-learning/tutorials/sentiment-analysis).
+
+## Dataset
+
+We'll be using a set of **Yelp reviews** as the input training data for this example. 
+
+For the specific ML training/predictions in this app (i.e. when using Model Builder), it helps to have a header for the data. We introduced the column header **ReviewText,** which holds the review itself, and **Sentiment,** which holds either a 0 to indicate negative sentiment or a 1 to indicate positive sentiment. You can view the resulting data set in [yelptrain.csv](./Resources/yelptrain.csv).
+
+You can [download the original Yelp data](https://archive.ics.uci.edu/ml/machine-learning-databases/00331/sentiment%20labelled%20sentences.zip) from the [UCI Sentiment Labeled Sentences Dataset]( https://archive.ics.uci.edu/ml/datasets/Sentiment+Labelled+Sentences). 
+
+## Solution
+
+We'll first train an ML model using ML.NET, and then we'll create a new application that uses both .NET for Apache Spark and ML.NET.
+
+> **Note:** All of the necessary files (trained ML model, .NET for Spark + ML.NET application code, training dataset) have been included in this project. You can follow the steps below to understand how the project/files were developed, recreate them yourself, and then adapt the steps to future applications.
+
+## ML.NET
```

Comment:
Instead of duplicating the model training example, can we have a link to the instructions in the other Sentiment example ?

Reply:
My thought-process here was allowing users to stay all in one doc to access the info they need. When docs start requiring me to view others, I can get confused/discouraged to have to keep navigating between different ones, and I'd prefer to have a flow all in the same place. It also avoids users wondering when to start/stop reading the other doc (i.e. if I link to other ML sentiment doc, they might read the model training section but wonder if they need to read the rest too).

That being said, I also totally get your POV. Do you think the benefits of a (potentially much) shorter readme outweigh a potentially more disjointed reading/learning experience?

Reply:
Also, whatever decision we make here, we can follow through the other sections you pointed out below.

Reply:
Only problem we will run into is if these instructions ever change, ie if ML.NET updates their tool and requires different steps.  We will just have to update both locations and review both files as well.

Maybe @imback82 or @rapoth can chime in.

Reply:
Hmm.  Since the two examples share so many similarities.   Can't we just combine the code into one example app?  First parameter can signify to run a batch or streaming analysis.   The udf, classes, and ml analysis code is the same in both examples.

Then you can combine the readme and have a section where the streaming and non streaming portions differ

Reply:
I definitely see what you're saying with the concern about versions changing. I think it could make more sense in that case to just link to the other relevant readmes (also, since this sample is a bit more advanced, users looking at it may already be familiar with the other samples or be fine with digging around a little).

Reply:
I thought about combining into one app, but I feel like it might:

- Get confusing and disrupt the flow (users who aren't familiar with the Spark flow yet might be confused what steps they should follow in general, i.e. reading into the DataFrame and displaying/formatting DataFrame is different for batch vs streaming)
- Be too long (I think we'd have to read in and finally display the DataFrames in different ways, and having both snippets would take up a decent amount of space, and the samples are already long)

Definitely open to your thoughts here.